sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobContainerAsyncClient.java

```diff
@@ -111,6 +111,7 @@
     private final CpkInfo customerProvidedKey; // only used to pass down to blob clients
     private final EncryptionScope encryptionScope; // only used to pass down to blob clients
     private final BlobContainerEncryptionScope blobContainerEncryptionScope;
+    private final String sasToken;
```

Comment:
We should cache `AzureSasCredential` instead. Otherwise this won't reflect if credential got rotated.

Reply:
That's a good point. I'll update that

Reply:
Are you ok with making the sasTokenCredential field protected?

Reply:
Another option would be to return the AzureSasCredential itself instead of returning a string. Then I think it should work with all the existing infrastructure and it would mean that we get the rotation stuff for free. I'm not sure what we think about returning credentials outright, though.

Reply:
yet another option is to have private reference in both base and subtype. I think either this or protected getter (not field).

Reply:
Yet another option is to put an accessor in the implementation package and use it internally. See:
https://github.com/Azure/azure-sdk-for-java/blob/f37bc26bf5a9acab8af040fe83ed02d7c4b88725/sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/util/AccountPropertiesHelper.java#L11
and 
https://github.com/Azure/azure-sdk-for-java/blob/f37bc26bf5a9acab8af040fe83ed02d7c4b88725/sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/administration/models/AccountProperties.java#L49-L63

Reply:
Seems a little verbose and convoluted for returning a property to me. Is there an objection to returning the credential?

Reply:
Or we can make the sasCredential a protected variable so that all its children have access to it. 

Protected variables on BlobClientBase are encryptionScope, accountName, containerName, blobName, serviceVersion
Private variables are snapshot, versionId, cpk.

It's not really clear to me what the distinction is between some of these values and why they're protected vs private. I don't see a big reason why we couldn't make the sasCredential protected, which would resolve this issue

Reply:
If we take a step back [the guideline](https://azure.github.io/azure-sdk/general_design.html#authentication) discourages exposing properties/accessors that would give access to authentication data in the clients. I recall we had similar requests to expose reference to credentials in the .NET SDK that has not been implemented and we've been usually able to provide alternative solutions to the customers. It might be worth understanding customer's scenario more, i.e. why do they need to get the sas token back.

Reply:
I asked the customer that opened the issue this PR specifically addresses. This was originally a feature request. However, some form of this is also a prerequisite for #25714 in which we need to be able to get the sas token from the async client to append it to the header value for the rename source

Reply:
Yeah, we need internalized version of this for the bug fix, i.e. without exposing public API. I'd suggest to focus on that now. 
Regarding the protected field on the base blob client, I think this should be ok. I wasn't aware of existing pattern.

Reply:
Agree with @kasobol-msft  here. We should avoid exposing credentials/tokens from the client. If it's needed for internal use, we can allow access to it without making it public.